vkubelet/pod.go

```diff
@@ -105,7 +105,21 @@ func (s *Server) startPodSynchronizer(ctx context.Context, id int) {
 	logger.Debug("Start pod synchronizer")
 
 	for event := range s.podCh {
-		s.syncPod(event.ctx, event.pod)
+		logger := log.G(event.ctx).WithField("podSyncronizer", id)
+		ctx := log.WithLogger(event.ctx, logger)
+
+		lockCtx, cancel := context.WithTimeout(ctx, 60*time.Second)
+		if err := s.lock(lockCtx, event.pod); err != nil {
```

Comment:
if we have multiple thread, lock cannot guarantee the event order. With the timeout, we might lost event if the operation accumulates.

Reply:
Yep, agreed. This does seem to make everything safer, but ultimately it's not a perfect solution.